src/components/routes/Booking/BookingOptions/SelectPaymentOption/SelectPaymentOption.tsx

```diff
@@ -67,18 +79,29 @@ class SelectPaymentOption extends React.Component<Props> {
   }
 
   handleSelectedCurrencyOption = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    this.setState({ currency: event.target.value });
+    const currency = event.target.value;
+    this.setState({ currency, conversionRateFromBee: undefined });
+    if (currency === Currency.DAI) {
+      const web3 = loadWeb3();
+      const beeQuote = this.props.booking.priceQuotes.find(q => q.currency === Currency.BEE);
+      if (beeQuote) {
+        const total = beeQuote.guestTotalAmount;
+        priceWithToken(web3.eth, currency, total)
+          .then(price => this.setState({ conversionRateFromBee: price / total }))
+          .catch(() => this.setState({ errorPricingToken: true }));
+      }
+    }
   };
 }
 
 export default SelectPaymentOption;
 
-function currencyOptions(currency: string | undefined, booking: Booking): React.ReactNode {
+function currencyOptions(currency: string | undefined, booking: Booking, fromBee?: (value: number) => (number | string)): React.ReactNode {
   switch (currency) {
     case Currency.BEE:
-      return <BookingOptionsCrypto booking={booking} currency={Currency.BEE} />;
+    case Currency.DAI:
```

Comment:
do the BEE and DAI cases need to return anything?

Reply:
I'm using [fallthrough](https://en.wikipedia.org/wiki/Switch_statement#Fallthrough) here (BEE/DAI/ETH all return the same component)

Reply:
interesting! TIL